File: app/src/at-vinhnguyen/java/asiantech/internship/summer/javacore/exercise18/CountTheNumberOfWords.java
package asiantech.internship.summer.javacore.exercise18;

import android.util.Log;
import java.util.Scanner;

public class CountTheNumberOfWords {

    private static final char SPACE = ' ';
    private static final char TAB = '\t';
    private static final char BREAK_LINE = '\n';
    private static final String TAG = CountTheNumberOfWords.class.getSimpleName();

    public static void main(String[] args) {
        Log.d(TAG, "Nhap chuoi can tim: ");
        String str = new Scanner(System.in, "UTF-8").nextLine();
        Log.d(TAG, "Số từ của chuỗi đã cho là:" + countWords(str) + "\n ");
    }

    private static int countWords(String input) {
        if (input == null) {
            return -1;
        }
        int count = 0;
        int size = input.length();
        boolean notCounted = true;
        for (int i = 0; i < size; i++) {
            if (input.charAt(i) != SPACE
                    && input.charAt(i) != TAB
                    && input.charAt(i) != BREAK_LINE) {
                if (notCounted) {
                    count++;
                    notCounted = false;
                }
            } else {
                notCounted = true;
            }
        }
        return count;
    }
}

File: app/src/at-vinhnguyen/java/asiantech/internship/summer/exercise18/CountTheNumberOfWords.kt
package exercise18

import java.util.*

object CountTheNumberOfWords {

    val SPACE = ' '
    val TAB = '\t'
    val BREAK_LINE = '\n'

    @JvmStatic
    fun main(args: Array<String>) {
        print("Nhap chuoi can tim: ")
        val str = Scanner(System.`in`).nextLine()
        print(
            "Số từ của chuỗi đã cho là: "
                    + countWords(str) + "\n"
        )
    }

    fun countWords(input: String?): Int {
        if (input == null) {
            return -1
        }
        var count = 0
        val size = input.length
        var notCounted = true
        for (i in 0 until size) {
            if (input[i] != SPACE && input[i] != TAB
                && input[i] != BREAK_LINE
            ) {
                if (notCounted) {
                    count++
                    notCounted = false
                }
            } else {
                notCounted = true
            }
        }
        return count
    }
}
